modules/structured_light/samples/sl_example.cpp

```diff
@@ -0,0 +1,149 @@
+        namedWindow("Display pattern", WINDOW_NORMAL);// Create a window for display.
+        setWindowProperty("Display pattern", WND_PROP_FULLSCREEN, WINDOW_FULLSCREEN);
+        imshow("Display pattern", pause);
+        waitKey();
+
+        if (isCaptureRefs)
+        {
+            for(auto i = 0; i < patterns_.size(); i++)
+            {
+                Mat frame;
+                cap >> frame;
+                if(frame.empty()) break; // end of video stream
+
+                namedWindow("Display pattern", WINDOW_NORMAL);// Create a window for display.
+                setWindowProperty("Display pattern", WND_PROP_FULLSCREEN, WINDOW_FULLSCREEN);
+                imshow("Display pattern", patterns_[i]);
+                waitKey();
+
+                Mat grayFrame;
+                cv::cvtColor(frame, grayFrame, COLOR_RGB2GRAY);
+                grayFrame.convertTo(grayFrame, CV_32FC1, 1.f/255);
+                refs_.push_back(grayFrame); //ADD ADDITIONAL SWITCH TO SELECT WHERE to SAVE
+
+            }
+        }
+
+        pause = Mat(projector_size, CV_64FC3, Scalar(0));
+        putText(pause, "Place the object", Point(projector_size.width/4, projector_size.height/4), FONT_HERSHEY_COMPLEX_SMALL, projector_size.width/400, Scalar(255,255,255), 2);
+        putText(pause, "Press any key when ready", Point(projector_size.width/4, projector_size.height/4+projector_size.height/15), FONT_HERSHEY_COMPLEX_SMALL, projector_size.width/400, Scalar(255,255,255), 2);
+        namedWindow("Display pattern", WINDOW_NORMAL);// Create a window for display.
+        setWindowProperty("Display pattern", WND_PROP_FULLSCREEN, WINDOW_FULLSCREEN);
+        imshow( "Display pattern", pause);
+        waitKey();
+
+        for(auto i = 0; i < patterns_.size(); i++)
+        {
+            Mat frame;
+            cap >> frame;
+            if( frame.empty() ) break; // end of video stream
+
+            namedWindow("Display pattern", WINDOW_NORMAL);// Create a window for display.
+            setWindowProperty("Display pattern", WND_PROP_FULLSCREEN, WINDOW_FULLSCREEN);
+            imshow( "Display pattern", patterns_[i]);
+            waitKey();
+
+            Mat grayFrame;
+            cv::cvtColor(frame, grayFrame, COLOR_RGB2GRAY);
+            grayFrame.convertTo(grayFrame, CV_32FC1, 1.f/255);
+            imgs_.push_back(grayFrame); //ADD ADDITIONAL SWITCH TO SELECT WHERE to SAVE
+
+        }
+
+        cap.release();
+    }
+}
+
+int main()
+{
+    int imgNum = 4;
+    cv::Size projector_size = cv::Size(512, 512);
+    string alg_type = "PCG";
+    vector<cv::Mat> patterns, refs, imgs;
+
+    structured_light::StructuredLightMono sl(projector_size, imgNum, 37, alg_type);
+
+    sl.generatePatterns(patterns, 0.3);
+
+    captureImages(patterns, refs, imgs);
+
+    string filename = "../images/calibration_result.xml";
```

Comment:
Avoid relative paths in samples (especially for fixed files).

---

> /images/

Should be placed in `samples/data` (or `samples/sl_data` because `calibration_result.xml` is ambiguous name)